### PR TITLE
Examples

### PR DESCRIPTION
**Pull request by tonyinme**
A set of examples is provided in the directory nalu-wind/examples.
Also, documentation for the examples is presented.

The examples are meant to ease the process of setting up Nalu input files for ABL and wind turbine simulations.
This is still a work in progress. 
The two main cases are working: Neutral ABL, turbine under uniform inflow.

The compiled version of the documentation is available [here](https://nalu-wind-tony.readthedocs.io/en/latest/source/examples/index.html).

### CODE REVIEW

File: docs/source/examples/introduction.rst [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+The :program:`wind-utils` repository can be downloaded as a submodule by running
+this command inside the :file:`nalu-wind/` repository::
+
+    git submodule init && git submodule update
+
+Now, :program:`wind-utils` can be compiled with :program:`Nalu`
+by enabling the compilation flag inside of
+the `CMakeLists.txt file <https://github.com/Exawind/nalu-wind/blob/master/CMakeLists.txt>`_::
+
+    option(ENABLE_WIND_UTILS "Build wind utils along with Nalu-Wind" ON)
```
[the remark this answers is not in the document]
Comment (tonyinme) [reply docs/source/examples/introduction.rst]: In that case, when exactly can this be done?
I wanted to do that on the nalu-wind-gcc.sh script on peregrine but am not sure where to add it.

File: examples/turbine_uniform_inflow/5MW_Baseline/ServoData/compile.sh [hunk 2]
```diff
@@ -0,0 +1 @@
+gfortran -w -Wline-truncation -ffree-line-length-none -fPIC -shared -o libDISCON.so DISCON.F90
```
[the remark this answers is not in the document]
Comment (tonyinme) [reply examples/turbine_uniform_inflow/5MW_Baseline/ServoData/compile.sh]: It is helpful to have the file with compilation instructions there, specially if the user intends to change something and compile the controller.
I have removed the .so from the version control.

File: examples/abl_neutral/template_input_files/ablNeutralEdge.yaml [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+            latitude: 45.0
+            earth_angular_velocity: 7.2921159e-5
+
+        - limiter:
+            pressure: no
+            velocity: no
+            enthalpy: yes
+
+        - peclet_function_form:
+            velocity: classic
```
[the remark this answers is not in the document]
Comment (tonyinme) [reply examples/abl_neutral/template_input_files/ablNeutralEdge.yaml]: Done

File: docs/source/examples/ABLexample.rst [hunk 4]
```diff
@@ -0,0 +1,69 @@
+.. _abl_neutral_example:
+
+Atmospheric Boundary Layer - Neutral
+------------------------------------
+
+This case is a large-eddy simulation of a neutral atmospheric boundary layer.
+The case uses periodic boundary conditions on the sides (east, west, north,
+south), a wall-model in the bottom wall and a stress free boundary condition at
+the top of the domain.
+A PI controller is used to drive the velocity at a given height.
```
[the remark this answers is not in the document]
Comment (tonyinme) [reply docs/source/examples/ABLexample.rst]: Done

File: examples/turbine_uniform_inflow/5MW_Baseline/Airfoils/Cylinder1.dat [hunk 5]
```diff
@@ -0,0 +1,58 @@
+! ------------ AirfoilInfo v1.01.x Input File ----------------------------------
```
[the remark this answers is not in the document]
Comment (tonyinme) [reply examples/turbine_uniform_inflow/5MW_Baseline/Airfoils/Cylinder1.dat]: If these files are small, I prefer to keep them here such that once the user clones the repository, it is easy to copy and paste them without moving the symlinks around the system.
Let me know if this makes sense.

File: docs/source/examples/ABLexample.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (tonyinme) [reply docs/source/examples/ABLexample.rst]: I realized that is it actually a proportional controller, not a PI. I have changed all this in the documentation.

**Comment by michaelasprague** [comment]
@tonyinme -- please move the Examples section under "User Manual"

**Comment by tonyinme** [comment]
This pull request is still on hold until I finish an example for a wind farm simulation with inflow-outflow.

**Comment by tonyinme** [comment]
Hey @sayerhs @michaelasprague @jrood-nrel @rcknaus 

This is ready for now.
I have added a wind farm case.
This case is a lot more difficult than the others and I will do some more cleanup later on.
However this is a working case for now which will do
1. precursor ABL
2. boundary data sampling from ABL
3. wind farm simulation

**Pull request merged**